packages/core/src/internal/execution/future-processor/helpers/network-interaction-execution.ts

```diff
@@ -153,12 +153,32 @@ export async function sendTransactionForOnchainInteraction(
       failedEstimateGasSimulationResult
     );
 
-    assertIgnitionInvariant(
-      decoded !== undefined,
-      "Expected failed simulation after having failed to estimate gas"
-    );
+    if (decoded !== undefined) {
+      return decoded;
+    }
+
+    // this is just for type inference
+    assertIgnitionInvariant(error instanceof Error, "Unexpected error type");
```

Comment:
```suggestion
    assertIgnitionInvariant(error instanceof Error, "Unexpected error type while resolve failed gas estimation");
```
I was wondering if we could do better here, so if it ever happens we have a hope of gleaning info from the error message. But rather than `json.stringify(...)` or rethrowing, a more specific error is enough.